crates/hir-def/src/data.rs

```diff
@@ -532,10 +538,11 @@ impl ExternCrateDeclData {
 pub struct ConstData {
     /// `None` for `const _: () = ();`
     pub name: Option<Name>,
-    pub type_ref: Interned<TypeRef>,
+    pub type_ref: TypeRefId,
```

Comment:
It is not (it is always the last actually because typerefs are allocated recursively), but the last is also easy to tell. I do wonder if we should make it so dependent on implicit behavior however, or maybe reserve the first to it somehow.

Reply:
So I tried this and this doesn't reduce memory usage at all. On some types this doesn't even shrink them because of padding. And this depends on subtle implementation detail, and, if we take this to the extreme (i.e. apply this for as many types as possible, e.g. impls) it introduces an order dependency between the type ref lowering which isn't clear from a glance and needs to be documented. As such, I'm inclined to not do this.